### ISSUE TITLE
`ScrollView`'s `centerContent` prop doesn't work

### ISSUE THREAD

**Issue by nandorojo** [necolas/react-native-web#2331]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the issue

As always, thanks for the incredible work of RNW. Onto the issue...

`<ScrollView centerContent={true} />` doesn't center the content. Here is a comparison of iOS and Web:

### iOS

<img width="1792" alt="Screen Shot 2022-07-06 at 5 04 19 PM" src="https://user-images.githubusercontent.com/13172299/177643412-41e29a59-0270-431c-b86c-2b9d297a6c49.png">

### Web

<img width="1792" alt="Screen Shot 2022-07-06 at 5 04 35 PM" src="https://user-images.githubusercontent.com/13172299/177643473-dce4ea48-b4be-48b7-8fd4-7cd2b4eafe4a.png">

### Expected behavior

`centerContent` should align content to the center.

### Steps to reproduce

1. Add a `<ScrollView centerContent>`
2. Put a view inside of it
3. Compare native and Web behavior

### Test case

https://snack.expo.dev/@beatgig/rnw-centercontent-broken

### Potential solution

Adjusting `contentContainerStyle` seems to have fixed it:

```tsx
import { View, StyleSheet, ScrollView } from 'react-native';

export default function App() {
  return (
    <ScrollView centerContent contentContainerStyle={styles.fix}>
      <View style={{ height: 50, backgroundColor: 'blue' }} />
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  fix: {
    justifyContent: 'center',
    flexGrow: 1,
  },
});
```

<img width="1792" alt="Screen Shot 2022-07-06 at 5 05 12 PM" src="https://user-images.githubusercontent.com/13172299/177643683-b143b672-739d-4f63-81f8-600a38034df3.png">

For additional context: if I didn't add `flexGrow: 1`, then the content wasn't properly scrollable when it overflowed the size of its container.

If that's the correct solution, please let me know and I'm happy to submit a PR. Thanks!

**Comment by necolas** [comment]
> If that's the correct solution, please let me know and I'm happy to submit a PR. Thanks!

Thanks! Looks good to me.

> …if I didn't add `flexGrow: 1`, then the content wasn't properly scrollable when it overflowed the size of its container.

I didn't understand this. Is this a different bug? Please can you elaborate?

**Comment by nandorojo** [comment]
Thanks for the quick reply.

> I didn't understand this. Is this a different bug? Please can you elaborate?

Yeah happy to elaborate, sorry for the confusion. It's not a separate bug. I just wanted to explain the reasoning behind the solution.

I first tried this:

```tsx
<ScrollView contentContainerStyle={{ justifyContent: 'center' }} />
```

This made the content stay up at the top.

<img width="1792" alt="Screen Shot 2022-07-06 at 5 49 03 PM" src="https://user-images.githubusercontent.com/13172299/177649382-a2c208e8-2c5d-46ec-83c4-11e44754752f.png">

This made the content go up to the top.

I then tried adding `flex: 1`:

```tsx
<ScrollView contentContainerStyle={{ justifyContent: 'center', flex: 1 }} />
```

That appeared to have fixed it.

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/13172299/177649531-a71bfdfe-f923-47d1-a1b4-11a59bf6cd32.png">

However, as the scrollable content got long, an issue arose:

<img width="1792" alt="Screen Shot 2022-07-06 at 5 51 02 PM" src="https://user-images.githubusercontent.com/13172299/177649645-2c5c2375-110f-4d5f-8135-6f99cd830046.png">

^ This screenshot shows the content scrolled all the way to the top. I can't go up more, so the content is essentially getting cut off. [Here's a video](https://www.loom.com/share/c680272709c74ba9bf47eef11969e276?from_recorder=1&focus_title=1) for clarity.

Once I changed `flex: 1` → `flexGrow: 1`, the cut-off content got fixed:

<img width="1792" alt="image" src="https://user-images.githubusercontent.com/13172299/177649956-841d48d0-6ed0-4960-b216-b5a84b91165b.png">

[Here is a video](https://www.loom.com/share/e60a1ba70b7640d999f8a000a83fe5e3) of the working behavior (`flexGrow: 1, justifyContent: 'center'`).

**Comment by necolas** [comment]
Worth noting that this prop is not supported on Android

**Comment by necolas** [comment]
Should be available in 0.18.6

**Comment by nandorojo** [comment]
Great, thanks so much for the quick fix.
